### DISCUSSION TITLE
Unable to delete files when using the File Reader smb protocol

### DISCUSSION THREAD

**Discussion by erandavid** [nextgenhealthcare/connect#4637] [Q&A]
Hi, 
We have a very simple channel with a File Reader source which scans a network folder using the smb protocol for suitable name files to process. 
The file reader gets the following error when trying to delete the file after it has been processed, and so is picked up again on the next poll:
ERROR (com.mirth.connect.connectors.file.FileReceiver:435): Error processing file in channel: com.mirth.connect.connectors.file.FileConnectorException: Error deleting file from [smb://...

The "Everyone" group has full control on this Windows share folder so it doesn't seems to be a permissions issue.

Did anyone encounter such an issue?

**Comment by jonbartels** [comment]
Is there more information in that stack trace? Further down the stack trace there may be a more specific cause that would help solve the issue. Anything in `mirth.log`?  (Use code tags when posting stack traces please!)

Is the file path shown in the `smb://` URL correct? Could this be something like a computed file path that isn't substituted properly?

**Reply by erandavid** [reply]
Here is the full stack trace from the Mirth log:

`[2021-07-20 05:05:15,142]  ERROR  (com.mirth.connect.connectors.file.FileReceiver:27): Error processing file in channel: 7fcaf4af-a0ba-4e1d-8f7e-e6d0738a1914
com.mirth.connect.connectors.file.FileConnectorException: Error moving file from [smb://wosapci/interfaces/HCM/Data/Medisparks//NTISH.dat] to [C:/SAPMessages/Archive/Success/NTISH_1626789915140.dat]
at com.mirth.connect.connectors.file.FileReceiver.processFile(FileReceiver.java:482)
at com.mirth.connect.connectors.file.FileReceiver.processFiles(FileReceiver.java:328)
at com.mirth.connect.connectors.file.FileReceiver.poll(FileReceiver.java:239)
at com.mirth.connect.donkey.server.channel.PollConnectorJob.execute(PollConnectorJob.java:49)
at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
`

Here's the channel File Reader setup:

![image](https://user-images.githubusercontent.com/7279820/126340452-30261259-d8b5-41f9-b351-6cd149835317.png)


The URL is correct (test using the 'Test Read' button)
There are no computed file paths.

**Reply by jonbartels** [reply]
Hmmm.

You checked the remote directory permissions already. Did you check the local side of the move operation? Does the user that the MC service runs as have access to the local directory? Does the *entire* local file path exist? 

The code for `com.mirth.connect.connectors.file.FileReceiver#renameFile` shows that the move operation can throw an exception but that exception is not logged, its swallowed and returns false. So if the code gets a more specific root-cause it won't show it to the caller.

**Reply by erandavid** [reply]
The MC service user is Local System.
It works fine if I change the source smb folder to a local folder
The smb folder has the group "Everyone" with a "Full control" permission included 
Note that the problem is just when deleting the file. Reading works ok.

**Reply by jonbartels** [reply]
That sounds like a permissions issue, would Windows logs show a problem on that end?

The last idea I have would be if SMB is sensitive to the double-slash before the file name. `smb://wosapci/interfaces/HCM/Data/Medisparks//NTISH.dat` 

I'll have to defer to another expert. Sorry I couldn't help solve the issue.

**Reply by erandavid** [reply]
I tried adding a double-slash before the file name but same results...
There is nothing on the Windows logs.
BTW: I am able to delete the file manually with the user provided in the smb dialog. That's another indication that it's not a permission issue.

**Reply by tonygermano** [reply]
It's only possible to move files to another location on the same file system.

**Reply by pacmano1** [reply]
Meaning you are trying to move from a remote file system to a local file system.   You can do that if you mount the file systems locally to the mirth server.

**Reply by tippenring** [reply]
I believe local system does not have access to network paths. Running the service as a different user with network access should be able to access network resources.

**Reply by tonygermano** [reply]
I almost brought that up, too, but the Windows service account doesn't matter since it is using the credentials added in the File Reader as shown in the screenshot.

Even though it's not relevant to this issue, Local System does have network access (it's Local Service that does not,) but it accesses the network as the Computer Account the machine used to join the domain, and there's a good chance that permission was not given to the server itself.
